CRM/Report/Form/Pledge/Pbnp.php

```diff
@@ -249,7 +274,11 @@ public function from() {
                         ON ({$this->_aliases['civicrm_pledge']}.contact_id =
                             {$this->_aliases['civicrm_contact']}.id)  AND
                             {$this->_aliases['civicrm_pledge']}.status_id IN ( {$statusIds} )
-             LEFT  JOIN civicrm_pledge_payment {$this->_aliases['civicrm_pledge_payment']}
+             LEFT  JOIN ";
```

Comment:
Just for the clarity if we you can move the LEFT JOIN to the below before '(SELECT ...' string

Overall the patch is working fine :)